src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1378,16 +1378,17 @@ private static bool HasPublicParameterlessConstructor(NamedTypeSymbol type, bool
         {
             Debug.Assert(type.TypeKind is TypeKind.Class or TypeKind.Struct);
 
-            // PROTOTYPE(req): Adjust for required members
+            bool hasAnyRequiredMembers = type.HasAnyRequiredMembers;
 
             foreach (var constructor in type.InstanceConstructors)
             {
                 if (constructor.ParameterCount == 0)
                 {
-                    return constructor.DeclaredAccessibility == Accessibility.Public;
+                    return constructor.DeclaredAccessibility == Accessibility.Public
+                        && (!hasAnyRequiredMembers || constructor.HasSetsRequiredMembers);
                 }
             }
-            return synthesizedIfMissing;
+            return synthesizedIfMissing && !hasAnyRequiredMembers;
```

Comment:
It seems like we're making the assumption that the synthesized parameterless constructor will never have 'HasSetsRequiredMembers'. Is that fine?

Reply:
I can't see how it wouldn't be.